x/gamm/pool-models/stableswap/stableswap_pool.go

```diff
@@ -0,0 +1,29 @@
+package stableswap
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+var _ types.PoolI = &Pool{}
+
+// NewStableswapPool returns a stableswap pool
+// Invariants that are assumed to be satisfied and not checked:
+// * len(initialLiquidity) = 2
+// * FutureGovernor is valid
```

Comment:
Is the idea that these two are checked by msg.ValidateBasic?

Reply:
Yes, but it looks like the initial liquidity validation [isn't implemented yet](https://github.com/osmosis-labs/osmosis/blob/e9b9804bcef341a2b22fa90d7722a2fadb738d7e/x/gamm/pool-models/stableswap/msgs.go#L46) – do you think this something that should be added in this PR or in a separate issue @ValarDragon?

Reply:
We can add that in a second PR

Reply:
Created an issue here: https://github.com/osmosis-labs/osmosis/issues/1364